packages/@react-aria/menu/src/useMenuItem.ts

```diff
@@ -131,27 +168,10 @@ export function useMenuItem<T>(props: AriaMenuItemProps, state: TreeState<T>, re
     ariaProps['aria-setsize'] = getItemCount(state.collection);
   }
 
-  let onKeyDown = (e: KeyboardEvent) => {
-    // Ignore repeating events, which may have started on the menu trigger before moving
-    // focus to the menu item. We want to wait for a second complete key press sequence.
-    if (e.repeat) {
-      return;
-    }
-
-    switch (e.key) {
-      case ' ':
-        if (!isDisabled && state.selectionManager.selectionMode === 'none' && closeOnSelect !== false && onClose) {
-          onClose();
-        }
-        break;
-      case 'Enter':
-        // The Enter key should always close on select, except if overridden.
-        if (!isDisabled && closeOnSelect !== false && onClose) {
-          onClose();
-        }
-        break;
-    }
-  };
+  if (isMenuDialogTrigger) {
+    ariaProps['aria-haspopup'] = 'dialog';
```

Comment:
If we get rid of this prop we'll need some way to know that this should be 'dialog' and not 'menu' (as for a submenu). 🤔

Reply:
yeah... i'm not sure what to do about this one

Reply:
we'll expose the aria-haspopup prop and set in spectrum